itest/litd_custom_channels_test.go

```diff
@@ -2766,3 +2771,457 @@ func testCustomChannelsFee(_ context.Context,
+	closeExpiryInfo := newCloseExpiryInfo(t.t, alice)
+
+	// With all of the HTLCs established, we'll now force close the channel with
+	// Alice.
+	t.Logf("Force close by Alice w/ HTLCs...")
+	daveChanPoint := &lnrpc.ChannelPoint{
+		OutputIndex: uint32(assetFundResp.OutputIndex),
+		FundingTxid: &lnrpc.ChannelPoint_FundingTxidStr{
+			FundingTxidStr: assetFundResp.Txid,
+		},
+	}
+	_, closeTxid, err := net.CloseChannel(alice, daveChanPoint, true)
+	require.NoError(t.t, err)
+
+	t.Logf("Channel closed! Mining blocks, close_txid=%v", closeTxid)
+
+	// Next, we'll mine a block which should will start the clock ticking on the
+	// relative timeout for the Alice, and Bob.
+	//
+	// After this next block, both of them can start to sweep.
+	//
+	// For Alice, she'll go to the second level, revealing her preimage in the
+	// process. She'll then need to wait for the relative timeout to expire
+	// before she can sweep her output.
+	//
+	// For Bob, since the remote party (Alice) closed, he can try to sweep right
+	// away after initial confirmation.
+	mineBlocks(t, net, 1, 1)
+
+	// After force closing, Bob should now have a transfer that tracks the force
+	// closed commitment transaction.
+	locateAssetTransfers(
+		t.t, bobTap, *closeTxid,
+	)
+
+	t.Logf("Settling Bob's hodl invoice")
+
+	// At this point, the commitment transaction has been mined, and we have 4
+	// total HTLCs on Alice's commitment transaction:
+	//
+	//  * 2x outgoing HTLCs to Alice to Bob
+	//  * 2x incoming HTLCs from Bob to Alice
+	//
+	// We'll left half the HTLCs timeout, while pulling the other half.
+	// To start, we'll signal Bob to settle one of his incoming HTLCs on Alice's
+	//
+	// commitment transaction. For him, this is a remote success spend, so
+	// there's no CSV delay other than the 1 CSV (carve out), and he can spend
+	// directly from the commitment transaction.
+	_, err = bob.InvoicesClient.SettleInvoice(
+		ctxb, &invoicesrpc.SettleInvoiceMsg{
+			Preimage: bobHodlInvoices[0].preimage[:],
+		},
+	)
+	require.NoError(t.t, err)
+
+	// With the invoice settled, we should now see Bob go to the chain with his
+	// direct spend from Alice's commitment transaction.
+	_, err = waitForNTxsInMempool(
+		net.Miner.Client, 1, time.Second*5,
+	)
+
+	// We'll mine an empty block to get the sweeper to tick.
+	mineBlocks(t, net, 1, 0)
+
+	// Next, we'll mine an additional block, this should allow Bob to sweep both
+	// his commitment output, as well as the incoming HTLC that we just settled
+	// above.
+	mineBlocks(t, net, 1, 1)
+
+	// At this point, we should have the next sweep transaction in the mempool:
+	// Bob's incoming HTLC sweep directly off the commitment transaction.
+	_, err = waitForNTxsInMempool(
+		net.Miner.Client, 1, time.Second*5,
+	)
+
+	// We'll now mine the next block, which should confirm Bob's HTLC sweep
+	// transaction.
+	mineBlocks(t, net, 1, 1)
+
+	// TODO(roasbeef): assert transfers
```

Comment:
Asserting that the correct proofs can be found in the universe would also be a great check.